src/utils/cosmosdb_model.ts

```diff
@@ -39,7 +42,7 @@ export const ResourceT = t.intersection([
     _ts: t.number
   }),
   BaseModel
-]) as t.Type<Resource>;
+]) as t.Type<Resource & { id: NonEmptyString }>; // this cast is needed to keep ResourceT in sync with Resource (try to remove a field from the decoder definition and you'll face an error)
```

Comment:
perhaps

```suggestion
]) as Omit<t.Type<Resource, "id"> & { id: NonEmptyString }>; // this cast is needed to keep ResourceT in sync with Resource (try to remove a field from the decoder definition and you'll face an error)
```

Reply:
Well, the `id` attribute would be override by the one on the right